core/unix/os.c

```diff
@@ -5684,20 +5684,22 @@ add_dr_env_vars(dcontext_t *dcontext, char *inject_library_path, const char *app
             "WARNING: execve env does NOT preload DynamoRIO, forcing it!\n");
         if (preload >= 0) {
             /* replace the existing preload */
+            char *dynamorio_library_path = get_dynamorio_library_path();
             sz = strlen(envp[preload]) + strlen(DYNAMORIO_PRELOAD_NAME) +
-                strlen(DYNAMORIO_LIBRARY_NAME) + 3;
+                strlen(dynamorio_library_path) + 3;
```

Comment:
This direct ref is relying on the call above?  I guess it's technically correct but it seems fragile, and the uses in the else below are not guaranteed to have called get_dynamorio_library_path() right?  Maybe have a local var and call at top of the routine?